consensus/core/src/test_dag.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use consensus_config::AuthorityIndex;
+use parking_lot::RwLock;
+
+use crate::{
+    block::{BlockRef, Round, TestBlock, VerifiedBlock},
+    context::Context,
+    dag_state::DagState,
+};
+
+/// Build a fully interconnected dag up to the specified round. This function
+/// starts building the dag from the specified [`start`] parameter or from
+/// genesis if none are specified up to and including the specified round [`stop`]
+/// parameter.
+pub fn build_dag(
+    context: Arc<Context>,
+    dag_state: Arc<RwLock<DagState>>,
+    start: Option<Vec<BlockRef>>,
+    stop: Round,
+) -> Vec<BlockRef> {
+    let mut ancestors = match start {
+        Some(start) => {
+            assert!(!start.is_empty());
+            assert_eq!(
+                start.iter().map(|x| x.round).max(),
+                start.iter().map(|x| x.round).min()
+            );
+            start
+        }
+        None => {
+            let (references, genesis): (Vec<_>, Vec<_>) = context
+                .committee
+                .authorities()
+                .map(|index| {
+                    let author_idx = index.0.value() as u32;
```

Comment:
I think there will be a dedicated method to created genesis blocks, although it is not yet merged. Leave a TODO to refactor?